array_test.go

```diff
@@ -1660,6 +1703,14 @@ func TestArray_Filter(t *testing.T) {
 		filteredArray.chain.assertNotFailed(t)
 	})
 
+	t.Run("nil func", func(t *testing.T) {
```

Comment:
Let's rename this to "invalid argument" and move to the end of the test, for consistency with other tests.

Reply:
Changed format to the consistency of other tests